client/app/lib/components/table/TanStackTableBuilder/customFlexRender.ts

```diff
@@ -0,0 +1,41 @@
+import { ReactNode } from 'react';
+import { Cell, flexRender, Header } from '@tanstack/react-table';
+
+import { RowSelector } from '../adapters';
+import { Data } from '../builder';
+
+import { ROW_SELECTOR_ID } from './columnsBuilder';
+
+export const customCellRender = <D extends Data>(
+  cell: Cell<D, unknown>,
+): ReactNode | RowSelector => {
+  const renderable = cell.column.columnDef.cell;
+  const context = cell.getContext();
+
+  if (cell.column.id === ROW_SELECTOR_ID) {
+    if (typeof renderable === 'function') {
+      return renderable(context) as RowSelector;
+    }
+
+    throw new Error('RowSelector renderer should be a function!');
+  }
+
+  return flexRender(renderable, context);
```

Comment:
I cant find much documentation on this, so I am wondering why not use flexRender for selector column?

Reply:
LOL Even I can't remember why again. I did remember figuring this out not by any documentations, but by reading TanStack Table's source code, and how `flexRender` is used. Definitely because otherwise, it throws an error(?)

I think it has to do with the fact that row selectors don't have actual data. In `columnsBuilder.ts:25`, row selectors are defined as [display columns](https://tanstack.com/table/v8/docs/guide/column-defs#:~:text=filtered%2C%20grouped%2C%20etc.-,Display%20Columns,-Display%20columns%20do), i.e., they don't have associated raw values. I think `flexRender` can't render these kinds of columns.